task/az/task.go

```diff
@@ -110,48 +110,59 @@ type Task struct {
 
 func (t *Task) Create(ctx context.Context) error {
 	logrus.Info("Creating resources...")
-	logrus.Info("[1/10] Creating ResourceGroup...")
-	if err := t.Resources.ResourceGroup.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[2/10] Creating StorageAccount...")
-	if err := t.Resources.StorageAccount.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[3/10] Creating BlobContainer...")
-	if err := t.Resources.BlobContainer.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[4/10] Creating Credentials...")
-	if err := t.DataSources.Credentials.Read(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[5/10] Creating VirtualNetwork...")
-	if err := t.Resources.VirtualNetwork.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[6/10] Creating SecurityGroup...")
-	if err := t.Resources.SecurityGroup.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[7/10] Creating Subnet...")
-	if err := t.Resources.Subnet.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[8/10] Creating VirtualMachineScaleSet...")
-	if err := t.Resources.VirtualMachineScaleSet.Create(ctx); err != nil {
-		return err
-	}
-	logrus.Info("[9/10] Uploading Directory...")
-	if t.Attributes.Environment.Directory != "" {
-		if err := t.Push(ctx, t.Attributes.Environment.Directory); err != nil {
+	// Define a list of steps to execute. If we adopt this pattern, we
+	// can declare a named type for the step definition.
+	steps := []struct {
```

Comment:
Is this change needed in this PR? I find it a bit out of the scope

Reply:
Agree, but why not follow up in technical debt issue? Just only to not loose the focus when reviewing it.

Reply:
it looks like https://github.com/iterative/terraform-provider-iterative/pull/632/commits/aaa06e9db2dd964870bc39749e8a8a609deb7c96 can pretty easily cherry-picked to another PR and rebased out?

Reply:
Good point, will do just that - thank you!